### DISCUSSION TITLE
Conflict caused by Tailwind CSS when imported as part of a React NPM package in NextJS app which has tailwind already installed

### DISCUSSION THREAD

**Discussion by christopherkindl** [tailwindlabs/tailwindcss#8446] [Help] [answered]
I created a custom React NPM package which is built on top of Tailwind CSS and includes it as a DevDependency:

main.css:
```
@tailwind base;
@tailwind components;
@tailwind utilities;
```

tailwind.config.js:
```
module.exports = {
    content: [
      "./src/**/*.{js,jsx,ts,tsx}",
    ],
    theme: {
      extend: {},
    },
    plugins: [],
  }
```

postcss.config.js 
```
module.exports = {
    plugins: {
        tailwindcss: {},
        autoprefixer: {},
    },
};
```

When the package is imported in a nextJS app which has tailwind already configured and installed, certain existing components of the nextJS app are not rendered anymore once the NPM package is installed and certain components of the package are called. However, the components coming from my react NPM package are rendered correctly.

Specifically, the sidebar suddenly disappears once components of the package are used on the site. The sidebar consists of components outside of the package but is styled using Tailwind CSS.

Before:
[![enter image description here][1]][1]

After:

[![Screenshot][2]][2]

Many thanks for any hints!

  [1]: https://i.stack.imgur.com/CmNvS.png
  [2]: https://i.stack.imgur.com/yS6VT.png

**Comment by christopherkindl** [comment, marked as answer]
Solved the issue by myself. Turns out you need to add `important: true` in tailwind.config.js.

```
module.exports = {
    content: [
        './pages/**/*.{js,ts,jsx,tsx,mdx}',
        './components/**/*.{js,ts,jsx,tsx,mdx}',
        './data/**/*.{js,ts,jsx,tsx,mdx}',
    ],
    theme: {
        extend: {},
    },
    plugins: [],
    important: true,
};
```

**Comment by joecacti** [comment]
Hi @christopherkindl - you might be able to help me out with an issue I'm running in to.

I built a ui library using tailwind/vue and now that I've updated to tailwind 3 the new JIT engine is removing all the classes not being used from the package.  

When I pull it into my apps and reference the package which sits in my node_modules and I start using a new class that was not used in the library it's not recognizing it.  Any ideas on how I can get those classes added once I start using the app to the apps css?

Thanks for any help you might be able to provide.
